### DISCUSSION TITLE
Validation Workflow and UI for users to verify QR_TYPE=URL for small QR code size

### DISCUSSION THREAD

**Discussion by ankitv89** [Sunbird-RC/community#826] [Show & Tell]
Hi,

Use Case:
To develop a small QR that could be used as a validation scan for certificates.

Solution Case:
1. QR_TYPE was defined as a URL and an endpoint was specified [Domain/certs/unique-id?Entity=EntityType
2. A UI to validate and display the certificate data (online validation)
3. The UI also displays the W3C-VC QR which the user can save and download.
4. Display Public Key that end users can use to do an offline validation
5. Share on Linkedin Option to Share the creds

Benefits:
1. Small QR Size, retain aesthetics of existing certificates
2. Single-step scan and verify
3. Ability to enhance the UI with multiple user journeys for users once they reach verification UI e.g. login to add this to digilocker etc.

Adoption Cases:
1. We have adopted this for Tarento's internal user case for create verifiable certificates
2. Aastrika Foundation is using this to generate Certificates for offline events

Screenshots:
![image](https://github.com/Sunbird-RC/community/assets/8529877/e0423172-4a02-4964-9849-0169224da139)

We would like to contribute this to Sunbird-rc. Please help us with that.
@rhwarrier  @thillaibirla

**Comment by srprasanna** [comment]
Hi @ankitv89 this is great news, let us connect on discord and we can help you get your changes into Sunbird-RC.

**Comment by surendrasinghs** [comment]
Thanks @ankitv89 , this is a good enhancement and much welcome! This is essentially de-coupling online verification from offline verification - assuming both QR (online & offline) are generated by default.

Please expand on how is this configurable depending on the use-case?

**Reply by ankitv89** [reply]
Hi,

Following are the configuration variables we allow out of the box:

[](# RC Endpoints

TOKEN_ENDPOINT="http://localhost:8080/auth/realms/sunbird-rc/protocol/openid-connect/token"
VERIFICATION_ENDPOINT="http://localhost:port/api/v1/verify"
CERTIFICATE_ENDPOINT="http://localhost:port/api/v1"

#Keycloak credentials
CLIENT_ID="client-id"
CLIENT_SECRET="secret"

#Image urls
LOGO="logo file"

#Server-details
PORT:3013

#Client-details
CERTIFICATE_VERIFICATION_URL= "URL"
ORG_ID="LinkedIn Prg Id"
LINKEDIN_BASE_URL="Linkedin Base url"

#UI Config
enableLinkedinShare=boolean
generatew3cvc=boolean

#Issuer Config - Footer
issuerName="Company name"
issuerEmail="support email"
issuerWebsite="Company Website")

The code is written in Express with SSR UI using Pug templates so further customization is easy as well. We have added default templates for the success and failure of validation

**Reply by ankitv89** [reply]
@surendrasinghs Hi, any update on this? We have this feature in the current sprint (along with the momentum), so changes can be incorporated quickly, if any.

**Reply by srprasanna** [reply]
Can we setup a demo on how this works and then move forward on submission?
